src/linker/utilities/docker_utils.py

```diff
@@ -55,20 +57,25 @@ def _load_image(client: DockerClient, image_path: Path) -> str:
 def _run_container(
     client: DockerClient,
     image_id: str,
-    input_data: List[Path],
+    step_inputs: List[StepInput],
     results_dir: Path,
     diagnostics_dir: Path,
     config: Optional[Dict[str, str]],
 ):
     logger.info(f"Running the container from image {image_id}")
-    volumes = {
+    volumes = {}
+    for step_input in step_inputs:
+        for outside_path, inside_path in step_input.bindings.items():
+            volumes[outside_path] = {"bind": f"{inside_path}", "mode": "ro"}
+    volumes.update(
         **{
-            str(dataset): {"bind": f"/input_data/main_input_{dataset.name}", "mode": "ro"}
-            for dataset in input_data
-        },
-        str(results_dir): {"bind": "/results", "mode": "rw"},
-        str(diagnostics_dir): {"bind": "/diagnostics", "mode": "rw"},
-    }
+            str(results_dir): {"bind": "/results", "mode": "rw"},
+            str(diagnostics_dir): {"bind": "/diagnostics", "mode": "rw"},
+        }
+    )
+    environment = {}
```

Comment:
Where is this variable used?

Reply:
oh good catch this is supposed to be passed into the environment call and is probably only working because I'm hiding the failure by putting all the data files in /main_input/ and /secondary_input/ directly